### PR TITLE
feat(python, rust): add `is_leap_year` to temporal expressions

### PR DESCRIPTION
**Pull request by borchero**
# Motivation

Pandas' `.dt` accessor provides a method [`is_leap_year`](https://pandas.pydata.org/docs/reference/api/pandas.DatetimeIndex.is_leap_year.html) which is sometimes useful. Although it is possible to compute leap year information pretty easily with polars even today, it results in quite ugly code. Hence, it would be nice to push the computation into the library itself.

This PR adds `is_leap_year` to the list of temporal expressions. I hope I added the new property in all the right places, it was a little difficult to navigate 😇

### CODE REVIEW

File: polars/polars-time/src/chunkedarray/datetime.rs [hunk 1]
```diff
@@ -41,6 +42,18 @@ pub trait DatetimeMethods: AsDatetime {
         cast_and_apply(self.as_datetime(), temporal::year)
     }
 
+    /// Extract year from underlying NaiveDate representation.
+    /// Returns whether the year is a leap year.
+    fn is_leap_year(&self) -> BooleanChunked {
+        let ca = self.year();
```
Comment (ritchie46) [inline polars/polars-time/src/chunkedarray/datetime.rs:48]: It's a bit wasteful to first allocate a whole year array. We can determine the year and leap year in one pass while iterating.

**Review by ritchie46 — COMMENTED** [review]
Good PR. It looks good. I have one remark about the actual implementation. This can be faster, see comments.

File: polars/polars-time/src/chunkedarray/datetime.rs (hunk 1 above)
Comment (borchero) [reply polars/polars-time/src/chunkedarray/datetime.rs]: Good point! I translated it into a kernel, hope that does it. I tried to limit repetitiveness but there was still a bit of code that I needed to write to get this to work.

**Comment by borchero** [comment]
(Unsure why the CI fails, quite sure this is unrelated to this PR)

**Comment by ritchie46** [comment]
Looks great! Thanks @borchero

**Pull request merged**